roles/matrix-synapse/templates/ext/appservice-discord/systemd/matrix-appservice-discord.service.j2

```diff
@@ -0,0 +1,29 @@
+[Unit]
+Description=Matrix Appservice Discord server
+After=docker.service
+Requires=docker.service
+Requires=matrix-synapse.service
+After=matrix-synapse.service
+
+[Service]
+Type=simple
+ExecStartPre=-/usr/bin/docker kill matrix-appservice-discord
+ExecStartPre=-/usr/bin/docker rm matrix-appservice-discord
+ExecStart=/usr/bin/docker run --rm --name matrix-appservice-discord \
+			--log-driver=none \
+			-e "UID={{ matrix_user_uid }}" -e "GID={{ matrix_user_gid }}" \
```

Comment:
Since this is based on the IRC bridge, it lacks some security improvements we've added to the other bridges (telegram, whatsapp):

- running containers a a regular user from the very beginning, instead of using `root` and letting the container drop privileges later (e.g. `--user={{ matrix_user_uid }}:{{ matrix_user_gid }}`, instead of `-e "UID={{ matrix_user_uid }}" -e "GID={{ matrix_user_gid }}"`)

- dropping capabilities (`--cap-drop=ALL`)

Looks like the Discord bridge is written in TypeScript.
This is different than the Telegram bridge (written in Python) and the Whatsapp bridge (written in Go).
It's also written by a different person.. So it may be that special tweaks are necessary to enable it (besides using `--user={{ matrix_user_uid }}:{{ matrix_user_gid }}` and `--cap-drop=ALL`), compared to what we've done for the other bridges.

We can do these after this pull request gets merged!

Reply:
made these changes. because the files in the matrix/appservice-discord/ directory were previously created as root via the old service, i had to change the ownership of those files in my setup. 
i believe this wont be an issue for new installs.
learned something about git along the way, to explain the mess of force-pushes below.
https://github.com/spantaleev/matrix-docker-ansible-deploy/pull/105/commits/c2834d222611db86ef9b3c1e4725ac07da0f35a3